api/v1beta1/azurecluster_default_test.go

```diff
@@ -683,6 +710,14 @@ func TestSubnetDefaults(t *testing.T) {
 								},
 								SecurityGroup: SecurityGroup{Name: "cluster-test-node-nsg"},
 								RouteTable:    RouteTable{Name: "cluster-test-node-routetable"},
+								NatGateway: NatGateway{
+									NatGatewayIP: PublicIPSpec{
+										Name: "",
+									},
+									NatGatewayClassSpec: NatGatewayClassSpec{
+										Name: "cluster-test-node-natgw",
+									},
+								},
```

Comment:
Can you add test case where the NatGateway Name is set but the subnet is using IPv6 and validate that the defaults are not being set? That way we can test the negative case of this [line](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3365/files#diff-016ca697ac4323c794d5dda09a16fd6152518acb4e3da6efef765fe5011a796bR134)

Reply:
Same concern with above, and the below case: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/81f88115494b6c390621641e6074a760877b4074/api/v1beta1/azurecluster_default_test.go#L571-L636 has covered the negative test for IPv6.